### PR TITLE
[feature] #2765: Generate synthetic genesis in `kagami`

### LINKED ISSUE #2765: [suggestion]: Generate syntatic genesis
### Feature request

Add functionality to `kagami` to generate syntactic genesis. 

### Motivation

Could be helpful for load testing and blockchain explorer.

### Who can help?

_No response_

### PR DESCRIPTION
**Pull request by Erigara**
Signed-off-by: Shanin Roman <shanin1000@yandex.ru>

### Description of the Change

Add option to `kagami` to generate synthetic genesis with specified number of domains, accounts, assets.  

### Issue

Closes #2765.

### Benefits

Possible to generate large genesis for load testing and blockchain explorer.

### Possible Drawbacks

None.

### Usage Examples or Tests *[optional]*

```bash
cargo run --bin kagami genesis --synthetic --domains 1 --accounts-per-domain 1 --assets-per-domain 1
```

### Alternate Designs *[optional]*

Could be separate script.

### CODE REVIEW

File: tools/kagami/src/main.rs [hunk 1]
```diff
@@ -175,16 +175,39 @@ mod genesis {
     use super::*;
 
     #[derive(StructOpt, Debug, Clone, Copy)]
-    pub struct Args;
+    pub struct Args {
+        /// If set generate synthetic genesis with specified number of domains, accounts and assets.
+        #[clap(short, long, action)]
```
Comment (appetrosyan) [inline tools/kagami/src/main.rs:180]: I'd imagine that it's much better to phrase this as an `enum`. So `genesis --default` is the default value, `genesis` becomes `genesis --default` and `genesis --synthetic` accepts the three args. 

Just a suggestion. Fix at will.

File: tools/kagami/src/main.rs [hunk 2]
```diff
@@ -211,6 +234,63 @@ mod genesis {
...
+                            ),
+                        );
+                        acc.push(mint);
+                    }
+                }
+            }
+            acc
+        }
+        .into_iter()
+        .map(Into::into);
```
Comment (appetrosyan) [inline tools/kagami/src/main.rs:284]: Good-enough for a start, but not quite representative of a real blockchain.

**Review by appetrosyan — APPROVED** [review]

File: tools/kagami/src/main.rs [hunk 3]
```diff
@@ -211,6 +234,63 @@ mod genesis {
...
+        assets_per_domain: u64,
+    ) -> color_eyre::Result<RawGenesisBlock> {
+        let mut builder = RawGenesisBlockBuilder::new();
+        for domain in 0..domains {
+            let mut domain_builder = builder.domain(format!("domain_{domain}").parse()?);
+
+            for account in 0..accounts_per_domain {
+                domain_builder = domain_builder.with_account(
+                    format!("account_{account}").parse()?,
+                    DEFAULT_PUBLIC_KEY.parse()?,
```
Comment (s8sato) [inline tools/kagami/src/main.rs:250]: It depends on the use of `synthetic`, but generating keys would be closer to the actual situation

**Review by s8sato — APPROVED** [review]
Did you intentionally renamed from `syntactic` to `synthetic`?

File: tools/kagami/src/main.rs (hunk 3 above)
Comment (appetrosyan) [reply tools/kagami/src/main.rs:250]: I agree. Please generate different keys. In the new account scheme two accounts shall not have the same primary key.

**Comment by Erigara** [comment]
> Did you intentionally renamed from `syntactic` to `synthetic`?

Yes, like [artificial](https://dictionary.cambridge.org/dictionary/english/synthetic).

File: tools/kagami/src/main.rs [hunk 4]
```diff
@@ -211,6 +234,63 @@ mod genesis {
...
+            for asset in 0..assets_per_domain {
+                domain_builder = domain_builder
+                    .with_asset(format!("asset_{asset}").parse()?, AssetValueType::Quantity);
+            }
+
+            builder = domain_builder.finish_domain();
+        }
+        let mut genesis = builder.build();
+
+        let mints = {
```
Comment (appetrosyan) [inline tools/kagami/src/main.rs:263]: These aren't strictly necessary, but since we have them, we should consider adding `Transfer`s in a separate issue.

**Pull request merged**